apps/extension/src/background/keyring/messages.ts

```diff
@@ -75,3 +80,120 @@ export class CheckPasswordMsg extends Message<boolean> {
     return CheckPasswordMsg.type();
   }
 }
+
+export class GenerateMnemonicMsg extends Message<string[]> {
+  public static type() {
+    return MessageTypes.GenerateMnemonic;
+  }
```

Comment:
I think this is a better approach! I was thinking something similar. This portion was inherited, with the exception of adding an enum with all possible values, though I don't think there is any use in `type()` being a function, and switching it to a field would be much simpler and cleaner (but this will need to be checked thoroughly to make sure all instances are caught)

Reply:
I started this issue to cover refactoring the `Message` type: #108 - I took a quick look at it while linting, and it will take a bit more work to consolidate all messages into a type and move `type()` to a field.